app/src/main/java/com/example/stromtracker/ui/geraete/auswertung/GeraeteAuswertungFragment.kt

```diff
@@ -0,0 +1,338 @@
+
+        if (data.isNotEmpty()) {
+            //WICHTIG! Bei Verwendung von mehr als einem Chart muss man beim erstellen / neu Zeichnen das aktuelle als aktiv markieren
+            APIlib.getInstance().setActiveAnyChartView(anyChartProduzent)
+
+            var pie = AnyChart.pie()
+            pie = initPieChart(pie)
+
+            pie.data(data)
+            pie.title(
+                "Gesamtproduktion "
+                        + roundTo2Decimal(produzentList.sumByDouble { geraete -> geraete.getJahresverbrauch() } * (-1))
+                        + " kWh"
+            )
+            pie.legend().title().text("Produzenten")
+
+            anyChartProduzent.setChart(pie)
+        } else
+            anyChartProduzent.visibility = View.GONE
+    }
+
+    fun reloadKategorieChart() {
+        val data: MutableList<DataEntry> = ArrayList()
+
+        //Geräte nach Kategorie Gruppieren, dann die Summe der Jahresverbrauche berechnen
+        //TODO Test mit mehreren Kategorien und mit Produzenten (diese sollten hier nicht angezeigt werden)
+        val sortedgeraete: Map<Int, List<Geraete>> =
+            verbraucherList.groupBy(keySelector = { it.getKategorieID() })
+        for (kat in sortedgeraete) {
+            val currGeraeteInKat = kat.value
+            var sum: Double = 0.0
+            for (geraet in currGeraeteInKat) {
+                sum += geraet.getJahresverbrauch()
+            }
+            if (sum > 0)
+                data.add(
+                    ValueDataEntry(
+                        kategorieList[currGeraeteInKat.first().getKategorieID() - 1].getName(), sum
+                    )
+                )
+        }
+
+        if(data.isNotEmpty()) {
+            //WICHTIG! Bei Verwendung von mehr als einem Chart muss man beim erstellen / neu Zeichnen das aktuelle als aktiv markieren
+            APIlib.getInstance().setActiveAnyChartView(anyChartKategorie)
+
+            var pie = AnyChart.pie()
+            pie = initPieChart(pie)
+
+            pie.data(data)
+
+            pie.title("Gesamtverbrauch Kategorien")
+            pie.legend().title().text("Kategorien")
+
+            anyChartKategorie.setChart(pie)
+        }
+        else
+            anyChartKategorie.visibility = View.GONE
+    }
+
+    fun reloadRaumChart() {
+        val data: MutableList<DataEntry> = ArrayList()
+
+        //Geräte nach Raum Gruppieren, dann die Summe der Jahresverbrauche berechnen
+        val sortedgeraete: Map<Int, List<Geraete>> =
+            verbraucherList.groupBy(keySelector = { it.getRaumID() })
+        for (raum in sortedgeraete) {
+            val currGeraeteInRaum = raum.value
+            var sum: Double = 0.0
+            for (geraet in currGeraeteInRaum) {
+                sum += geraet.getJahresverbrauch()
+            }
+            if (sum > 0)
+                data.add(
+                    ValueDataEntry(
+                        raumList[currGeraeteInRaum.first().getRaumID() - 1].getName(), sum
```

Comment:
Das funktioniert so nicht, wir können nicht annehmen, dass die RaumID = Position in RaumListe ist (z.b wegen Löschen, und da wir nur Räume betrachten sollten, die in unserem Haushalt sind)

Reply:
Wird geändert, sobald current_Haushalt gemerget ist. Hab es bei Trello hinzugefügt.